pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcResolver.java

```diff
@@ -0,0 +1,364 @@
+                    break;
+                case DATE:
+                    value = result.getDate(colName);
+                    break;
+                case TIMESTAMP:
+                    value = result.getTimestamp(colName);
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Field type '" + DataType.get(oneField.type).toString() + "' (column '" + column.toString() + "') is not supported");
+            }
+
+            oneField.val = value;
+            fields.add(oneField);
+        }
+        return fields;
+    }
+
+    /**
+     * setFields() implementation
+     *
+     * @return OneRow with the data field containing a List<OneField>
+     * OneFields are not reordered before being passed to Accessor; at the
+     * moment, there is no way to correct the order of the fields if it is not.
+     * In practice, the 'record' provided is always ordered the right way.
+     *
+     * @throws UnsupportedOperationException if field of some type is not supported
+     */
+    @Override
+    public OneRow setFields(List<OneField> record) throws UnsupportedOperationException, ParseException {
+        int column_index = 0;
+        for (OneField oneField : record) {
+            ColumnDescriptor column = columns.get(column_index);
+            if (
+                LOG.isDebugEnabled() &&
+                DataType.get(column.columnTypeCode()) != DataType.get(oneField.type)
+            ) {
+                LOG.warn("The provided tuple of data may be disordered. Datatype of column with descriptor '" + column.toString() + "' must be '" + DataType.get(column.columnTypeCode()).toString() + "', but actual is '" + DataType.get(oneField.type).toString() + "'");
+            }
+
+            // Check that data type is supported
+            switch (DataType.get(oneField.type)) {
+                case BOOLEAN:
+                case INTEGER:
+                case FLOAT8:
+                case REAL:
+                case BIGINT:
+                case SMALLINT:
+                case NUMERIC:
+                case VARCHAR:
+                case BPCHAR:
+                case TEXT:
+                case BYTEA:
+                case TIMESTAMP:
+                case DATE:
+                    break;
+                default:
+                    throw new UnsupportedOperationException("Field type '" + DataType.get(oneField.type).toString() + "' (column '" + column.toString() + "') is not supported");
+            }
+
+            if (LOG.isDebugEnabled()) {
+                if (DataType.get(oneField.type) == DataType.BYTEA) {
+                    LOG.debug("OneField content (conversion from BYTEA): '" + new String((byte[])oneField.val) + "'");
+                }
+            }
+
+            // Convert TEXT columns into native data types
+            if ((oneField.val != null) && (DataType.get(oneField.type) == DataType.TEXT) && (DataType.get(column.columnTypeCode()) != DataType.TEXT)) {

```

Comment:
This bit failed for us when we had a timestamp or date field that had a NULL value. The type reported by GPDBWritable was not being converted from TEXT to timestamp, because conversion of types would only apply if the value were not NULL. Even if the value is null, we want oneField's type to be the column type code and not string. 

Here's the snippet that worked for us:
```
            if (LOG.isDebugEnabled()) {
                if (DataType.get(oneField.type) == DataType.BYTEA) {
                    LOG.debug("OneField content (conversion from BYTEA): '" + new String((byte[])oneField.val) + "'");
                }
            }
            // convert values of non TEXT types represented as text by the GPDBWritable serializer into native types
            if ((DataType.get(oneField.type) == DataType.TEXT) && (DataType.get(column.columnTypeCode()) != DataType.TEXT)) {
                // convert type information
                oneField.type = column.columnTypeCode();

                // Convert values if they are present (not null)
                if ((oneField.val != null)) {
                    String rawVal = (String)oneField.val;
                    if (LOG.isDebugEnabled()) {
                        LOG.debug("OneField content (conversion from TEXT): '" + rawVal + "'");
                    }
        ...
```
and remove
```
oneField.type = column.columnTypeCode();
```
from the bottom of the `if` statement.